### PR TITLE
[Mono][RISC-V] emit mul inst instead of emulated mul

### PR DESCRIPTION
**Pull request by Xinlong-Wu**
- emit mul inst instead of emulated mul
- try handle exceptions
- emit inst idiv
- process ArgOnStack arg at prologue 
- fix logic emit_imm

### CODE REVIEW

**Comment by Xinlong-Wu** [comment]
btw, what should I do if I want to set up a CI for RISC-V Arch?

**Comment by am11** [comment]
> btw, what should I do if I want to set up a CI for RISC-V Arch?

Wait for https://github.com/dotnet/runtime/pull/85443 😉

File: src/mono/mono/metadata/jit-icall-reg.h [hunk 1]
```diff
@@ -341,6 +341,7 @@ MONO_JIT_ICALL (ves_icall_thread_finish_async_abort) \
 MONO_JIT_ICALL (mono_marshal_lookup_pinvoke) \
 MONO_JIT_ICALL (mono_gsharedvt_constrained_call_fast) \
 MONO_JIT_ICALL (mono_dummy_runtime_init_callback) \
+MONO_JIT_ICALL (mono_riscv_throw_exception) \
 	\
```
Comment (vargaz) [inline src/mono/mono/metadata/jit-icall-reg.h:345]: This is not needed.

**Review by vargaz — APPROVED** [review]

**Comment by am11** [comment]
@Xinlong-Wu, with this PR, helloworld app works using `dotnet` (tested https://github.com/dotnet/runtime/pull/84470#discussion_r1176563648). 👍

I then tested this sample: https://github.com/dotnet/samples/tree/7b5da1b14bb3efc24d97e104ddbe5d3bb88f8cbc/csharp/getting-started/console-webapiclient, it throws from lowering:

```sh
$ /artifacts/bin/testhost/net8.0-linux-Debug-riscv64/dotnet bin/Release/net8.0/console-webapiclient.dll

unable to lowering following IR: long_mul_imm R45 <- R42
* Assertion: should not be reached at /runtime/src/mono/mono/mini/mini-riscv.c:2709
```
(aside: maybe reword `unable to lowering following IR` message as `unable to lower the following IR`)

**Comment by Xinlong-Wu** [comment]
> @Xinlong-Wu, with this PR, helloworld app works using `dotnet` (tested [#84470 (comment)](https://github.com/dotnet/runtime/pull/84470#discussion_r1176563648)). 👍
> 
> I then tested this sample: https://github.com/dotnet/samples/tree/7b5da1b14bb3efc24d97e104ddbe5d3bb88f8cbc/csharp/getting-started/console-webapiclient, it throws from lowering:
> 
> ```shell
> $ /artifacts/bin/testhost/net8.0-linux-Debug-riscv64/dotnet bin/Release/net8.0/console-webapiclient.dll
> 
> unable to lowering following IR: long_mul_imm R45 <- R42
> * Assertion: should not be reached at /runtime/src/mono/mono/mini/mini-riscv.c:2709
> ```
> 
> (aside: maybe reword `unable to lowering following IR` message as `unable to lower the following IR`)

yeah, cuz the port of Mono was not been completed. I think the IR `long_mul_imm` will be lowered in next pr

**Comment by Xinlong-Wu** [comment]
ping, can this pr been merged?
